TarponSimulator2017/Core/Physics.cs

```diff
@@ -1,11 +1,137 @@
 ﻿using System;
+using Microsoft.Xna.Framework;
+using Tarpon.Utils;
 
 namespace Tarpon.Core
 {
+
 	public class Physics
 	{
-		public Physics ()
+		
+
+		public float FrictionForce{ get; private set; }
+		public float AccelerationForce{ get; private set; }
+		public float TurnSpeed{ get; private set; }
+		public float MaxTurnAngle{ get; private set; }
+
+		public Vector2 Orientation { get; private set; }
+
+		/** Est recalculé à chaque tour **/
+		public Vector2 Acceleration { get; private set; }
+
+
+		public Vector2 Speed { get; private set; }
+
+
+		public Vector2 CentralPosition { get; private set; }
+
+
+		/**Le point de l'objet où la physique s'applique**/
+		public Vector2 ApplicationPoint { get; private set; }
+
+
+		public float TurnAngleSpeed { get; private set; }
+
+
+		public float TurnAngle { get; private set; }
+
+
+		int elapsedTime; // in milliseconds
+
+
+		public Physics(){
+		}
+
+		public Physics (float FrictionForce, float AccelerationForce, float TurnSpeed, float MaxTurnAngle,
+			int AbscisseDepart, int OrdonneeDepart	)
 		{
+			this.FrictionForce = FrictionForce;
+			this.AccelerationForce = AccelerationForce;
+			this.TurnSpeed = TurnSpeed;
+			this.MaxTurnAngle = MaxTurnAngle;
+
+			CentralPosition = new Vector2 (AbscisseDepart, OrdonneeDepart);
+			ApplicationPoint = new Vector2 (AbscisseDepart, OrdonneeDepart);
+
+			Orientation = new Vector2(0, -1);
+
+
+
+		}
+
+		public float OrientiationFloat()
+		{
+			return (float)Math.Atan2(Orientation.X, -Orientation.Y);
+		}
+
+		public void Accelerate()
+		{
+			Acceleration = Orientation * AccelerationForce;
+			Console.WriteLine ("Accelerate triggered");
+		}
+
+		public void Turn(Utils.Direction d)
+		{
+			TurnAngleSpeed = (int)d * TurnSpeed;
+			Console.WriteLine ("Turn triggered");
+		}
+
+		public void ComputeMovement(int now)
+		{
+			Console.WriteLine ("ComputeMovement triggered");	
+			elapsedTime = now;
```

Comment:
I know that I'm the only responsible for this, but we can really get rid of it. Just replace:

```csharp
public void ComputeMovement(int now)
```

by

```csharp
public void ComputeMovement(int elapsedTime)
```